src/parameterFiles.ts

```diff
@@ -0,0 +1,507 @@
+// ----------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation.  All rights reserved.
+// ----------------------------------------------------------------------------
+
+import * as assert from 'assert';
+import * as fse from 'fs-extra';
+import * as path from 'path';
+import { commands, ConfigurationTarget, MessageItem, TextDocument, Uri, ViewColumn, window, workspace } from 'vscode';
+import { callWithTelemetryAndErrorHandling, DialogResponses, IActionContext, IAzureQuickPickItem, UserCancelledError } from 'vscode-azureextensionui';
+import { configKeys, configPrefix, globalStateKeys, isWin32 } from './constants';
+import { DeploymentTemplate } from './DeploymentTemplate';
+import { ext } from './extensionVariables';
+import { containsParamsSchema } from './schemas';
+
+const readAtMostBytesToFindParamsSchema = 4 * 1024;
+const currentMessage = "Current";
+const similarFilenameMessage = "Similar filename";
+const howToMessage = `You can manually associate a parameter file with this template at any time by clicking "Select Parameter File..." in the status bar or the editor context menu.`;
+
+const _filesToIgnoreThisSession: Set<string> = new Set<string>();
```

Comment:
>Set<string>(); [](start = 51, length = 14)

Nit: not sure if we should use case-insensitive file name set here on Windows (here and in other places we use sets of filenames)

Reply:
I didn't worry about it here because it's this instance only and likely to match. Also, consequences of being wrong are minor.  Will add comments.